File: backend/internal/domain/entities/process_stat.go
package entities

import "fmt"

type ProcessStat struct {
	Pid           int32   `json:"pid"`
	Name          string  `json:"name"`
	Status        string  `json:"status"`
	CPUPercent    float64 `json:"cpu_percent"`
	MemoryPercent float32 `json:"memory_percent"`
}

func NewProcessStat(pid int32, name, status string, cpuPercent float64, memoryPercent float32) *ProcessStat {
	return &ProcessStat{
		Pid:           pid,
		Name:          name,
		Status:        status,
		CPUPercent:    cpuPercent,
		MemoryPercent: memoryPercent,
	}
}

func (p *ProcessStat) ToString() string {
	return fmt.Sprintf(
		"PID: %d, Name: %s, Status: %s, CPU Percent: %.2f, Memory Percent: %.2f",
		p.Pid, p.Name, p.Status, p.CPUPercent, p.MemoryPercent)
}

File: frontend/lib/models/process_stat.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'process_stat.freezed.dart';
part 'process_stat.g.dart';

@freezed
class ProcessStat with _$ProcessStat {
  const factory ProcessStat({
    required int pid,
    required String name,
    required String status,
    @JsonKey(name: 'cpu_percent') required double cpuPercent,
    @JsonKey(name: 'memory_percent') required double memoryPercent,
  }) = _ProcessStat;

  factory ProcessStat.fromJson(Map<String, dynamic> json) =>
      _$ProcessStatFromJson(json);
}
